### PR TITLE
Soft deprecate webrick and add some details as to why.

### LINKED ISSUE #12: webrick dependency necessary?
This repo was extracted from rack for v3, which previously didn't explicitly depend on webrick.

Webrick was dropped as a bundled gem from ruby 3 and the default web server for Rails is puma.

The other supported servers aren't depended upon and it would be nice if webrick is the same when I'm not actually using it.

### PR DESCRIPTION
**Pull request by ioquatix**
@jeremyevans I have copied your advice from https://github.com/rack/rackup/issues/12#issuecomment-2452746091 and moved `webrick` to a development dependency.

Fixes #12.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-`rackup` provides a command line interface for running a Rack-compatible application.
+`rackup` provides a command line interface for running a Rack-compatible application. It also provides a generic interface for starting a `rack`-compatible server: `Rackup::Handler`. It is not designed for production use.
 
 [![Development Status](https://github.com/rack/rackup/workflows/Test/badge.svg)](https://github.com/rack/rackup/actions?workflow=Test)
 
 ## Installation
 
 ``` bash
-$ gem install rackup
+-- For Falcon
```
Comment (jeremyevans) [inline readme.md:10]: Considering relative popularity, Puma should be listed first.

**Review by jeremyevans — APPROVED** [review]
Looks good.  Thank you for working on this.

File: readme.md (hunk 1 above)
Comment (ioquatix) [reply readme.md:10]: Haha, I did it by alphabetical order. Are you okay with that or do you still want me to change it?

Comment (jeremyevans) [reply readme.md:10]: I recommend the change, but I approved without it, so it's up to you.

Comment (ioquatix) [reply readme.md:10]: Okay, I don't have a strong opinion, so I'll change it.

**Comment by jeremy** [comment]
If rackup itself is soft-deprecated, how about leaving the webrick dep in place and unwinding the compatibility snafu?

Allow it to continue behaving just as it did before, and advise folks to move off Rackup instead.

If Rackup does carry on, make the webrick dep change in a major release (3.0).

**Comment by ioquatix** [comment]
I don’t have a strong opinion about this except that some of the things that broke already looked kind of broken to me.

I don’t think we should encourage further usage of webrick either. So if we did what you suggest and users upgrade to 3, they would be in the same boat? In other words the best solution for people right now is to pin the version until downstream dependencies are fixed? Or just accelerate the deployment and release of those fixes?

**Pull request merged**